src/core/agent/services/identifierService.ts

```diff
@@ -410,15 +417,45 @@ class IdentifierService extends AgentService {
           isPending,
         });
 
+        this.props.eventEmitter.emit<IdentifierStateChangedEvent>({
```

Comment:
I missed the change a few lines up. Why do we have `displayName: groupId,`? It should be the display name, not the groupId